openmdao/core/problem.py

```diff
@@ -759,90 +760,167 @@ def _compute_total_derivs(self, of=None, wrt=None, return_format='flat_dict',
                         ikey = old_input_list[icount]
                         totals[okey][ikey] = -approx_jac[output_name, input_name]
 
+            return totals
+
         # Solve for derivs using linear solver.
-        else:
+        voi_lists = defaultdict(list)
+        v2rhs_name = {}
```

Comment:
To clarify, I was referring to `v2rhs_name` specifically, but more comments would be welcome for `voi_lists` and `old_input_list` as well